### PR TITLE
Nn 2599 cell search part 1

### PR DESCRIPTION
**Pull request by Nimphal**
Adds tables, data and repositories. Service and resource to follow in a separate PR.

### CODE REVIEW

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    public static class PK implements Serializable {
+
+        @Column(name = "LIVING_UNIT_ID", insertable = false, updatable = false, nullable = false)
+        private Long livingUnitId;
+
+        @Column(name = "DESCRIPTION", nullable = false)
+        private String description;
+
+        @Column(name = "AGY_LOC_ID", nullable = false)
+        private String agencyLocationId;
```
Comment (Nimphal) [inline src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java:43]: No table primary key, so being a bit creative.

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnitProfile.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+        @Column(name = "DESCRIPTION")
+        private String description;
+
+        @Column(name = "AGY_LOC_ID")
+        private String agencyLocationId;
+
+        @Column(name = "PROFILE_ID")
+        private Long profileId;
+
```
Comment (Nimphal) [inline src/main/java/net/syscon/elite/repository/jpa/model/LivingUnitProfile.java:47]: Same as above.

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    @AllArgsConstructor
+    public static class PK implements Serializable {
+
+        @Column(name = "LIVING_UNIT_ID", insertable = false, updatable = false, nullable = false)
+        private Long livingUnitId;
+
+        @Column(name = "DESCRIPTION", nullable = false)
+        private String description;
+
+        @Column(name = "AGY_LOC_ID", nullable = false)
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java:42]: We have the models in place to support this already:

```
    @ManyToOne(optional = false)
    @JoinColumn(name = "AGY_LOC_ID", nullable = false)
    private AgencyLocation agencyLocationId;

```

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    private String lowestLevelFlag;
+
+    @Column(name = "REACH_OPER_CAPACITY_FLAG")
+    private String reachedOperationalCapacityFlag;
+
+    @Column(name = "NO_OF_OCCUPANT")
+    private Integer noOfOccupants;
+
+    public boolean isActive() {
+        return activeFlag.equals("Y");
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java:142]: Not really a big deal (as `activeFlag` should not be null) but I find it better to do something like: `"Y".equals(activeFlag)` this helps with preventing nullpointers.

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    public boolean isActiveCell() {
+        return isActive() && livingUnitType != null && livingUnitType.equals("CELL");
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java:146]: Same hear as above (avoid the need for the null check):

`return isActive() && "CELL".equals(livingUnitType);`

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnitProfile.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @Id
+    @Column(name = "LIVING_UNIT_ID", nullable = false)
+    private Long livingUnitId;
+
+    @Id
+    @Column(name = "AGY_LOC_ID", nullable = false)
+    private String agencyLocationId;
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/repository/jpa/model/LivingUnitProfile.java:56]: See agency comment above.  We have an object to model this.

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+import static org.hibernate.annotations.NotFoundAction.IGNORE;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@IdClass(LivingUnit.PK.class)
+@Table(name = "LIVING_UNITS")
+public class LivingUnit {
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java:30]: As a general comment do you need to map all of the fields here?   I tend to only map what is needed.

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java (hunk 3 above)
Comment (Nimphal) [reply src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java:42]: In this case the agency location column is part of the composite primary key, so I don't know if it's possible to use the model instead. Also don't need the agency information as part of the object, purely used for filtering.

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnitProfile.java (hunk 6 above)
Comment (Nimphal) [reply src/main/java/net/syscon/elite/repository/jpa/model/LivingUnitProfile.java:56]: Same as above.

File: src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java (hunk 3 above)
Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/repository/jpa/model/LivingUnit.java:42]: Just realised I put this on the wrong line anyway. Meant for line 53 which I think would be OK.  Not matter.

**Pull request merged**